liquidctl/driver/asetek_pro.py

```diff
@@ -0,0 +1,271 @@
+"""liquidctl drivers for sixth generation Asetek 690LC liquid coolers.
+
+Supported devices:
+
+- Corsair H100i PRO RGB, H115i PRO RGB or H150i PRO RGB (Experimental)
+
+Copyright (C) 2018–2020  Jonas Malaco and contributors
+
+Incorporates or uses as reference work by Kristóf Jakab, Sean Nelson
+and Chris Griffith.
+
+SPDX-License-Identifier: GPL-3.0-or-later
+"""
+
+import logging
+
+import usb
+
+from liquidctl.driver.asetek import Modern690Lc
+from liquidctl.driver.usb import UsbDriver
+from liquidctl.error import NotSupportedByDevice
+from liquidctl.keyval import RuntimeStorage
+from liquidctl.util import clamp
+
+
+LOGGER = logging.getLogger(__name__)
+
+_MAX_PROFILE_POINTS = 7
+_CRITICAL_TEMPERATURE = 60
+_HIGH_TEMPERATURE = 45
+_READ_LENGTH = 32
+_READ_TIMEOUT = 2000
+_WRITE_TIMEOUT = 2000
+
+_PRO_WRITE_ENDPOINT = 0x1
+_PRO_READ_ENDPOINT = 0x81
+_PRO_CMD_READ_FIRMWARE = 0xaa
+_PRO_CMD_READ_AIO_TEMP = 0xa9
+_PRO_CMD_READ_FAN_SPEED = 0x41
+_PRO_CMD_WRITE_FAN_SPEED = 0x40
+_PRO_CMD_READ_PUMP_SPEED = 0x31
+_PRO_CMD_READ_PUMP_MODE = 0x33
+_PRO_CMD_WRITE_PUMP_MODE = 0x32
+_PRO_PUMP_MODES = [
+    'Quiet',
+    'Balanced',
+    'Performance'
+    ]
+
+_PRO_SPEEDS = {
+    'rainbow': [
+        0x30, # slow
```

Comment:
Are other speed values allowed or are these specific values the only ones allowed?

Reply:
The values in this data structure are the only ones I believe are allowed.